### DISCUSSION TITLE
replace value for a key in a json document

### DISCUSSION THREAD

**Discussion by setop** [ast-grep/ast-grep#1306] [Q&A]
I saw ast-grep supports json language.

I've tried to set a rule to replace the value of a given key but [failed](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Impzb24iLCJxdWVyeSI6ImNvbnNvbGUubG9nKCRNQVRDSCkiLCJyZXdyaXRlIjoibG9nZ1hYWFhYWFhYWGVyLmxvZygkTUFUQ0gpIiwiY29uZmlnIjoiIyBZQU1MIFJ1bGUgaXMgbW9yZSBwb3dlcmZ1bCFcbiMgaHR0cHM6Ly9hc3QtZ3JlcC5naXRodWIuaW8vZ3VpZGUvcnVsZS1jb25maWcuaHRtbCNydWxlXG5cbmlkOiB2ZXJzaW9uanNvblxubGFuZ3VhZ2U6IGpzb25cbnJ1bGU6XG4gIHBhdHRlcm46IHxcbiAgICBcInZlcnNpb25cIjogJFxuZml4OiB8LVxuICAgIFwidmVyc2lvblwiOiBcIjIuMFwiIiwic291cmNlIjoie1xuICBcIm5hbWVcIjogXCJmdW5wcm9qXCIsXG4gIFwidmVyc2lvblwiOiBcIjEuMFwiLFxuICBcInByaXZhdGVcIjogZmFsc2UsXG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcIkBhcG9sbG8vY2xpZW50XCI6IFwiXjMuMy4xNlwiLFxuICAgIFwiQG9ydWdhLXVpL29ydWdhLW5leHRcIjogXCIwLjguMTJcIixcbiAgICBcInZlcnNpb25cIjogXCJmb29cIlxuICB9ICBcbn0ifQ==)

Any hint ?

**Comment by HerringtonDarkholme** [comment]
Use playground https://ast-grep.github.io/playground.html and 

<img width="890" alt="image" src="https://github.com/ast-grep/ast-grep/assets/2883231/602f46b7-1b1d-4c0f-a48e-04be937a6315">

pattern object https://ast-grep.github.io/guide/rule-config/atomic-rule.html#pattern-object

You can also ask the question on stackoverflow https://stackoverflow.com/questions/tagged/ast-grep

**Reply by setop** [reply]
And ... this is what I did and I get the same error as what you screenshot-ed.

Thank you for the RTFM.

What is the point of the discussion section if users can't help each other ?

**Reply by HerringtonDarkholme** [reply]
You are not reading the doc and just ask the question without showing your effort. You can ask the question on  https://stackoverflow.com/questions/tagged/ast-grep and see how to ask question.

I hope you can do your homework properly next time before asking.


https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Impzb24iLCJxdWVyeSI6ImZvbygkJCRBLCBiLCAkJCRDKSIsInJld3JpdGUiOiIiLCJjb25maWciOiJydWxlOlxuICBwYXR0ZXJuOiBcbiAgICBjb250ZXh0OiAne1widmVyc2lvblwiOiBcIiRWRVJcIiB9J1xuICAgIHNlbGVjdG9yOiBwYWlyIiwic291cmNlIjoie1xuICAgIFwidmVyc2lvblwiOiBcInZlclwiXG59In0=

**Comment by setop** [comment]
I'm a maintainer of quite a big software and operate some websites based of it.
Like most of developers I used to write patches using diff/patch.
And like most of developers I swear when a patch fails because some chars have changed in the context.
I think ast-grep is really brillant in its approach : syntax is a much more reliable way to express the location of the code to change, compare to text.

So I wanted to skill up with ast-grep to change my patching system. I read most of the documentation, some parts many times.
My feeling as a newcomer is that this tool is very powerful but the examples provided are too contrived and do not illustrate any non basic case.

Beside I find the playground a great idea to learn and share. That is what I used to show what I tried for this question.
If the answer was trivial and the adoption was high, I guess someone would have guide me.
Instead the question was immediately closed with RTFM/GotoSO. This feels not welcoming.

BTW, thanks for the hint. I managed to write [my rule](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Impzb24iLCJxdWVyeSI6ImNvbnNvbGUubG9nKCRNQVRDSCkiLCJyZXdyaXRlIjoibG9nZ1hYWFhYWFhYWGVyLmxvZygkTUFUQ0gpIiwiY29uZmlnIjoiIyBZQU1MIFJ1bGUgaXMgbW9yZSBwb3dlcmZ1bCFcbiMgaHR0cHM6Ly9hc3QtZ3JlcC5naXRodWIuaW8vZ3VpZGUvcnVsZS1jb25maWcuaHRtbCNydWxlXG5cbmlkOiB2ZXJzaW9uanNvblxubGFuZ3VhZ2U6IGpzb25cbnJ1bGU6XG4gIHBhdHRlcm46XG4gICAgc2VsZWN0b3I6IHBhaXJcbiAgICBjb250ZXh0OiAne1widmVyc2lvblwiOiBcIiRBXCInXG5maXg6IHwtXG4gICAgXCJ2ZXJzaW9uXCI6IFwiMi4wXCIiLCJzb3VyY2UiOiJ7XG4gIFwibmFtZVwiOiBcImZ1bnByb2pcIixcbiAgXCJ2ZXJzaW9uXCI6IFwiMS4wXCIsXG4gIFwicHJpdmF0ZVwiOiBmYWxzZSxcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwiQGFwb2xsby9jbGllbnRcIjogXCJeMy4zLjE2XCIsXG4gICAgXCJAb3J1Z2EtdWkvb3J1Z2EtbmV4dFwiOiBcIjAuOC4xMlwiLFxuICAgIFwidmVyc2lvblwiOiBcImZvb1wiXG4gIH0gIFxufSJ9). But honestly, even if I was not far, I would never have guessed that I have to put the open curly brace and the double quotes on the value to make the pattern to match.

**Reply by HerringtonDarkholme** [reply]
Thank your detailed context.
Please note that this project is a hobby project, and our community is still small with a low adoption rate. While we strive to be helpful, support is not guaranteed unless the question demonstrates proof of work. And we welcome well-formed and detailed questions that can help build the community norm for asking questions.

I’ve made the documentation as helpful as possible to ensure the project is self-serving and scalable for various people with different backgrounds. For instance, the pattern object is highlighted twice in the pattern documentation, which is likely one of the first documents people read when learning about this project. https://ast-grep.github.io/guide/pattern-syntax.html
<img width="697" alt="image" src="https://github.com/user-attachments/assets/8d1d5d62-1463-47b2-ae4b-c9abd1a02816">
<img width="737" alt="image" src="https://github.com/user-attachments/assets/9d33b4bb-ee89-43f2-8d50-47a0ed145435">
I also added a new faq page for recurrent questions about wrong patterns. https://ast-grep.github.io/advanced/faq.html

And finally, your contributions are welcome! You can also contribute to our catalog with non-trivial, real-world examples at ast-grep catalog. https://ast-grep.github.io/catalog/
